lib/src/node/value/calculation.dart

```diff
@@ -0,0 +1,128 @@
+    jsThrow(JsError('Argument must be unquoted SassString'));
+  }
+}
+
+/// The JavaScript `SassCalculation` class.
+final JSClass calculationClass = () {
+  var jsClass =
+      createJSClass('sass.SassCalculation', (Object self, [Object? _]) {
+    jsThrow(JsError("new sass.SassCalculation() isn't allowed"));
+  });
+
+  jsClass.defineStaticMethods({
+    'calc': (Object argument) {
+      isCalculationValue(argument);
+      return SassCalculation.unsimplified('calc', [argument]);
+    },
+    'min': (Object arguments) {
+      var argList = jsToDartList(arguments).cast<Object>();
+      argList.forEach(isCalculationValue);
+      return SassCalculation.unsimplified('min', argList);
+    },
+    'max': (Object arguments) {
+      var argList = jsToDartList(arguments).cast<Object>();
+      argList.forEach(isCalculationValue);
+      return SassCalculation.unsimplified('max', argList);
+    },
+    'clamp': (Object min, [Object? value, Object? max]) {
+      if (value == null && max != null) {
+        jsThrow(JsError('`value` is undefined and `max` is not undefined'));
+      }
+      if ((value == null || max == null) &&
+          !(min is SassString && min.text.startsWith('var(')) &&
+          !(value is SassString && value.text.startsWith('var('))) {
+        jsThrow(JsError(
+            '`value` or `max` is undefined and neither `min` nor `value` is a SassString that begins with "var("'));
+      }
+      [min, value, max].whereNotNull().forEach(isCalculationValue);
+      return SassCalculation.unsimplified(
+          'clamp', [min, value, max].whereNotNull());
+    },
+  });
+
+  jsClass.defineMethods({
+    'assertCalculation': (SassCalculation self, [String? name]) => self,
+  });
+
+  jsClass.defineGetters({
+    // The `name` getter is included by default by `createJSClass`
+    'arguments': (SassCalculation self) => ImmutableList(self.arguments),
+  });
+
+  getJSClass(SassCalculation.unsimplified('calc', [SassNumber(1)]))
+      .injectSuperclass(jsClass);
+  return jsClass;
+}();
+
+/// The JavaScript CalculationOperation class
+final JSClass calculationOperationClass = () {
+  var jsClass = createJSClass('sass.CalculationOperation',
+      (Object self, String strOperator, Object left, Object right) {
+    var operator = CalculationOperator.values
+        .firstWhereOrNull((value) => value.operator == strOperator);
+    if (operator == null) {
+      jsThrow(JsError('Invalid operator: $strOperator'));
+    }
+    isCalculationValue(left, checkUnquoted: false);
+    isCalculationValue(right, checkUnquoted: false);
+    return SassCalculation.operateInternal(operator, left, right,
+        inMinMax: false, simplify: false);
+  });
+
+  jsClass.defineMethods({
+    'equals': (CalculationOperation self, Object other) => self == other,
+    'hashCode': (CalculationOperation self) => self.hashCode,
+  });
+
+  jsClass.defineGetters({
+    'operator': (CalculationOperation self) => self.operator.operator,
```

Comment:
...unfortunately the call stack error is back for `CalculationOperation.operator`. Because in JS we want this to be a string and not the enum it is in Dart, we have to define this getter explicitly to get the nested `operator.operator`. Trying to access this property is what is causing the current test failures:

```
Uncaught RangeError: Maximum call stack size exceeded
    at Rti._generalAsCheckImplementation [as _as] (/workspaces/dart-sass/build/npm/sass.dart.js:2682:34)
    at _callDartFunctionFastCaptureThis (/workspaces/dart-sass/build/npm/sass.dart.js:7468:26)
    at CalculationOperation0.<anonymous> (/workspaces/dart-sass/build/npm/sass.dart.js:7455:18)
    at calculationOperationClass__closure2.call$1 (/workspaces/dart-sass/build/npm/sass.dart.js:82775:51)
    at Object.Primitives_applyFunction (/workspaces/dart-sass/build/npm/sass.dart.js:1202:30)
    at Object.Function_apply (/workspaces/dart-sass/build/npm/sass.dart.js:5464:16)
    at _callDartFunctionFastCaptureThis (/workspaces/dart-sass/build/npm/sass.dart.js:7472:16)
```

I visited each of those and couldn't find anything that's obviously wrong. This is `sass.dart.js:82775` for reference

```js
A.calculationOperationClass__closure2.prototype = {
  call$1($self) {
    return type$.CalculationOperation._as($self).operator.operator;
  },
  $signature: 372
};
```

In general, this seems to be a problem with redefining getters in the JS class that are already defined in the Dart class... but it's a pattern that's been used extensively in other files like [`color.dart`](https://github.com/oddbird/dart-sass/blob/13b4f337d7e57856512f8d13cc9bae04ca9e54c7/lib/src/node/value/color.dart#L56-L64), so I don't know why it's failing here 😕

Reply:
Thanks, that did the trick